Tasks/AzureResourceGroupDeployment/main.ts

```diff
@@ -6,25 +6,29 @@ import virtualMachine = require("./operations/VirtualMachine");
 import resourceGroup = require("./operations/ResourceGroup");
 
 function run(): Promise<void> {
-    var taskParameters = new deployAzureRG.AzureRGTaskParameters();
-    var resourceGroupOperationsController = new resourceGroup.ResourceGroup(taskParameters);
-    var virtualMachineOperation = new virtualMachine.VirtualMachine(taskParameters);
-    switch (taskParameters.action) {
-        case "Create Or Update Resource Group":
-            return resourceGroupOperationsController.createOrUpdateResourceGroup();
-        case "DeleteRG":
-            return resourceGroupOperationsController.deleteResourceGroup();
-        case "Select Resource Group":
-            return resourceGroupOperationsController.selectResourceGroup();
-        case "Start":
-        case "Stop":
-        case "Restart":
-        case "Delete":
-        case "StopWithDeallocate":
-            return virtualMachineOperation.execute();
-        default:
-            throw tl.loc("InvalidAction", taskParameters.action);
-    }
+    return new deployAzureRG.AzureRGTaskParameters().getAzureRGTaskParameters().then((taskParameters) => {
+        var resourceGroupOperationsController = new resourceGroup.ResourceGroup(taskParameters);
+        var virtualMachineOperation = new virtualMachine.VirtualMachine(taskParameters);
+        switch (taskParameters.action) {
+            case "Create Or Update Resource Group":
+                return resourceGroupOperationsController.createOrUpdateResourceGroup();
+            case "DeleteRG":
+                return resourceGroupOperationsController.deleteResourceGroup();
+            case "Select Resource Group":
+                return resourceGroupOperationsController.selectResourceGroup();
+            case "Start":
+            case "Stop":
+            case "Restart":
+            case "Delete":
+            case "StopWithDeallocate":
+                return virtualMachineOperation.execute();
+            default:
+                throw tl.loc("InvalidAction", taskParameters.action);
+        }
+    }).catch((error) => {
```

Comment:
Directly we can't call a async method synchronously from a sync method. To achieve this all operations are being performed after promise has been resolved.